dockers/docker-sonic-telemetry/certificate_rollover_checker

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+"""
+certificate_rollover_checker
+
+This script will be leveraged to periodically check whether the certificate and private key
+files of streaming telemetry were rolled over by dSMS service or not. The streaming telemetry
+container will be restarted if the certificate and private key are rolled over by dSMS service
+and then updated by ACMS agent running in ACMS container.
+"""
+
+import os
+import signal
+import sys
+import syslog
+import time
+
+from swsscommon import swsscommon
+
+CERTIFICATE_CHECKING_INTERVAL_SECS = 3600
+
+
+def get_file_last_mod_time(file_path):
+    """Gets the last modification time of a specific file.  Args:
+        file_path: A string represents the file path.
+
+    Returns:
+        last_mod_time: A float number in seconds represents the last moditification time of file
+                       since epoch.
+    """
+    last_mod_time = 0.0
+
+    try:
+        last_mod_time = os.path.getmtime(file_path)
+    except OSError as error:
+        syslog.syslog(syslog.LOG_ERR,
+                      "Could not get last modification time of the file and error message is '{}'.".format(error))
+        sys.exit(1)
+
+    return last_mod_time
+
+
+def restart_streaming_telemetry():
+    """Restarts the streaming telemetry container by terminating the root process.
+
+    Args:
+        None
+
+    Returns:
+        None
+    """
+    root_process_pid = os.getppid()
+    syslog.syslog(syslog.LOG_INFO,
+                  "Restarting streaming telemetry service by terminating the process with pid: '{}'".format(root_process_pid))
+    os.kill(root_process_pid, signal.SIGTERM)
```

Comment:
>SIGTERM [](http://example.com/codeflow?start=37&length=7)

The modern convention is to use sighup (ref: [https://stackoverflow.com/a/28327659/2514803](https://stackoverflow.com/a/28327659/2514803) ).

The benefit is not to explicitly terminate the other process and trigger critical process monitor alerts.

You may need to add the sighup handler in sonic-telemetry if not supported.

Reply:
I think we can probably use the `supervisorctl restart telemetry` command to only restart the streaming telemetry server process once the secrets were rotated. This can avoid triggerring the critical process alerts.

Reply:
Restarting is using `SIGTERM` and `SIGKILL` internally. One big concern is graceful shutdown. Considering the client will often fetch large amount of data, graceful shutdown will make client easier.

Reply:
I got the point. We can do this by either sending the signal `SIGKILL` or the command `supervisorctl restart telemetry`. However, our main focus is how we can do some cleanup before gracefully stopping the telemetry server process and disconnecting with gNMI client side.